consul/share/plugins/consul/30_generic_methods/variable_from_consul.cf

```diff
@@ -0,0 +1,74 @@
+#####################################################################################
+# Copyright 2018 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+#
+# @name Variable from Consul
+# @description Gets a key-value dictionnary from Consul given the secret path
+#
+# @documentation To use the generated variable, you must use the form `${variable_prefix.variable_name}` with each name replaced with the parameters of this method.
+#
+# Access to the Consul has to be configured on each agent in /var/rudder/plugin-resources/consul.json. A sample config file is provided in /opt/rudder/share/plugins/consul/sample_consul.json
+#
+# This generic method will report as performing a repair when it fetches a key, and an error when it fails to do so.
+#
+# @parameter variable_prefix The prefix of the variable name
+# @parameter variable_name   The variable to define, the full name will be variable_prefix.variable_name
+# @parameter path            Consul kv path
+#
+# @class_prefix Variable_from_consul
+# @class_parameter path
+
+bundle agent variable_from_consul(variable_prefix, variable_name, path)
+{
+  vars:
+      "old_class_prefix" string => canonify("variable_from_consul_${path}");
+      "promisers" slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
+      "class_prefix" string => canonify(join("_", "promisers"));
+      "args" slist => {"${variable_prefix}", "${variable_name}", "${path}"};
+
+    pass1::
+      "${variable_prefix}.${variable_name}" string => execresult('/usr/bin/curl -s -H "X-Consul-Token: ${consul.config[token]}" http://${consul.config[server_addr]}/v1/kv/${pa$
```

Comment:
The end of this line seems broken (probably a copy/paste issue)

Reply:
oups! j'ai pas fait attention